x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsResultProcessor.java

```diff
@@ -117,7 +117,6 @@ public void process(AnalyticsProcess<AnalyticsResult> process) {
                 completeResultsProgress();
             }
             completionLatch.countDown();
-            process.consumeAndCloseOutputStream();
```

Comment:
Here is the offending code piece. Latch down was counted down before we close output stream.